File: ios/Sources/Detonator/Module/FullScreenModule/FullScreenModule.swift
import UIKit

class FullScreenModule: Module {
    override func register() -> Void {
        detonator.addRequestClass(key: "com.iconshot.detonator.fullscreen/open", requestClass: FullScreenOpenRequest.self)
        detonator.addRequestClass(key: "com.iconshot.detonator.fullscreen/close", requestClass: FullScreenCloseRequest.self)
    }
    
    static var parent: UIView? = nil
    static var view: ViewLayout? = nil
    static var index: Int? = nil
    static var layoutParams: LayoutParams? = nil
    static var fullScreenViewController: FullScreenViewController? = nil
}

File: android/Detonator/src/main/java/com/iconshot/detonator/module/fullscreen/FullScreenModule.java
package com.iconshot.detonator.module.fullscreen;

import android.view.ViewGroup;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.layout.ViewLayout;
import com.iconshot.detonator.module.Module;

public class FullScreenModule extends Module {
    public FullScreenModule(Detonator detonator) {
        super(detonator);
    }

    @Override
    public void setUp() {
        detonator.setRequestClass("com.iconshot.detonator.fullscreen::open", FullScreenOpenRequest.class);
        detonator.setRequestClass("com.iconshot.detonator.fullscreen::close", FullScreenCloseRequest.class);
    }

    public static ViewGroup parent = null;
    public static ViewLayout view = null;
    public static Integer index = null;
    public static ViewLayout.LayoutParams layoutParams = null;
    public static ViewLayout fullScreenView = null;
}
